Oracle.md

```diff
@@ -0,0 +1,198 @@
+
+At maturity time, if the BTC/USD rate is less than $999999, the oracle signs each of the digits representing the outcome value using the digits' corresponding R value.
+For example, if the rate at maturity time is $20000, the oracle will sign the digits `0`, `2`, `0`, `0`, `0`, `0`.
+
+If the rate is greater than $999999, the oracle signs the digits `9`, `9`, `9`, `9`, `9`, `9`.
+
+### Serialization and signing of outcome values
+
+Every outcome value should be encoded as [a UTF-8 string with NFC normalization](./Messaging.md#Fundamental-types), hashed using sha256 before finally being signed by an oracle.
+UTF-8 is chosen as being a widely supported and easy to implement encoding format.
+
+For numerical outcomes represented in bases greater than 10, each digit should be converted to base 10 before being encoded (note that base 10 numbers in UTF-8 take values in the 0x0030-0x0039 range).
+This helps preventing any confusion about the capitalization of letters or the introduction of non-standard characters.
+
+### Serialization of event descriptors
+
+Event descriptors should be serialized using [TLV format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) as described bellow.
+
+#### Version 0 `enum_event_descriptor`
+
+1. type: 55302 (`enum_event_descriptor_v0`)
+2. data:
+   * [`u16`:`num_outcomes`]
+   * [`string`:`outcome_1`]
+   * ...
+   * [`string`:`outcome_n`]
+
+This type of event descriptor is a simple enumeration where the value `n` is the number of outcomes in the event.
+
+Note that `outcome_i` is the outcome value itself and not its hash that will be signed by the oracle.
+
+#### Version 0 `range_event_descriptor`
+
+1. type: 55304 (`range_event_descriptor_v0`)
+2. data:
+   * [`int32`:`start`]
+   * [`u32`:`count`]
+   * [`u16`:`step`]
+   * [`string`:`unit`]
+   * [`int32`:`precision`]
+
+#### Version 0 `digit_decomposition_event_descriptor`
+
+1. type: 55306 (`digit_decomposition_event_descriptor_v0`)
+2. data:
+   * [`bigsize`:`base`]
+   * [`bool`:`is_signed`]
+   * [`string`:`unit`]
+   * [`int32`:`precision`]
+   * [`uint16`:`nb_digits`]
+
+### Oracle events
+
+For users to be able to create DLCs based on a given event, they also need to obtain information about the oracle and the time at which it plans on releasing a signature over the event outcome.
+Oracle events contain such information, which includes:
+* the oracle public key,
+* the nonce(s) that will be used to sign the event outcome(s)
+* the earliest time (UTC) at which it plans on releasing a signature over the event outcome, in epoch seconds,
+* the event descriptor,
+* the event ID which can be a name or categorization associated with the event by the oracle.
+
+The TLV serialization for oracle events is as follow:
+
+#### Version 0 `oracle_event`
+
+1. type: 55330 (`oracle_event_v0`)
+2. data:
+   * [`x_point`:`oracle_public_key`]
```

Comment:
It doesn't make sense have the oracle public key here. It creates the hazard of someone using the public key here to validate the signature. They must already know the oracle's public key and must not decode `oracle_event` until they have validated the signature. It would be good to stress this in the document.

Reply:
Where would you put it then? The reason for having it in this message is for when you get the data from a third party (not directly from the oracle). In that case you need to have some id for the oracle, and the public key seems the best to me. Am I misunderstanding what you mean?

Reply:
Applications can decide where to put it. In some applications they may decide that oracles are identified by their public key and send that key along with the announcement. In other application they may decide oracles are identified by something else.  In other cases their may be clients that are bound only to one oracle so they don't need an identifier. Regardless of whether the public key is sent, there's no need for the oracle to put this in the announcement data.

Does this make sense? If you guys are intent on using the public key as an oracle identifier in the p2p protocol then perhaps just slot the public key right next to this data in https://github.com/discreetlogcontracts/dlcspecs/blob/master/Messaging.md

Reply:
I think for now we can just move this pubkey to the announcement TLV?

Reply:
If we are removing the pub key from the tlv we should add a recommendation somewhere that the event id should have some reference to the oracle (whether it be the pubkey, a website, or just something like @OutcomeObserver).

Reply:
I think it is fair to say that an event TLV is worthless without the signature, and that the public key should come with the signature (in the announcement for example). I'm not sure what you mean by remove from TLV as we would add it to the announcement?

Reply:
Moved the public key to the oracle announcement as @nkohen suggested.